libraries/Microsoft.Bot.Builder.LanguageGeneration/TemplateEngine.cs

```diff
@@ -51,22 +51,25 @@ public TemplateEngine AddFiles(IEnumerable<string> filePaths, ImportResolverDele
             {
                 importResolver = importResolver ?? ((id) =>
                  {
-                    // import paths are in resource files which can be executed on multiple OS environments
-                    // Call GetOsPath() to map / & \ in importPath -> OSPath
-                    var importPath = GetOsPath(id);
-                    if (!Path.IsPathRooted(importPath))
+                     // import paths are in resource files which can be executed on multiple OS environments
+                     // Call GetOsPath() to map / & \ in importPath -> OSPath
+                     var importPath = GetOsPath(id);
+                     if (!Path.IsPathRooted(importPath))
                      {
-                        // get full path for importPath relative to path which is doing the import.
-                        importPath = Path.GetFullPath(Path.Combine(Path.GetDirectoryName(filePath), id));
+                         // get full path for importPath relative to path which is doing the import.
+                         importPath = Path.GetFullPath(Path.Combine(Path.GetDirectoryName(filePath), id));
                      }
-                    return (File.ReadAllText(importPath), importPath);
+                     return (File.ReadAllText(importPath), importPath);
                  });
 
                 var fullPath = Path.GetFullPath(filePath);
-                this.AddText(File.ReadAllText(fullPath), fullPath, importResolver);
+                var parsedTemplates = this.ParseContent(File.ReadAllText(fullPath), fullPath, importResolver);
+                this.Templates.AddRange(parsedTemplates);
             }
 
-            RunStaticCheck(Templates);
+            this.Templates = this.RemoveDuplicatedTemplates(this.Templates);
```

Comment:
Dedup is simple and good enough, or you just inline the linq here, and add comments.

Reply:
Oh, I see, the reason you do it here, is because you want to share this capability also in AddText. I'm not sure we want to do that for text also.